### DISCUSSION TITLE
How can I limit the number of jobs when Redis memory is almost full?

### DISCUSSION THREAD

**Discussion by iosifnicolae2** [OptimalBits/bull#1976] [Q&A]
I need to implement this mechanism because Bull can't process existing jobs when it does not have enough memory, so the Queue will be stuck until I allocate more memory.

**Comment by manast** [comment]
Normally you set a maximum amount of completed/failed jobs to keep so that memory does not grow indefinitely, check ```removeOnComplete``` and ```removeOnFail```: https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queueadd

**Reply by iosifnicolae2** [reply]
Yes, but we can't control the number of ingested packages, so I need to be able to limit the number of jobs pushed to Redis.

**Reply by manast** [reply]
ah ok, I understand what you mean. Although not perfect you could implement a limit on the number of jobs waiting in the queue and throw an error if it is full. You can get the number of jobs waiting in a queue with https://github.com/OptimalBits/bull/blob/develop/REFERENCE.md#queuegetwaitingcount (if you have a lot of delayed jobs you may want to also get that count)
